src/Transform/feature2Texture.js

```diff
@@ -95,37 +102,55 @@ function drawFeature(ctx, feature, origin, scale, extent, style = {}) {
     }
 }
 
-export default {
-    createTextureFromFeature(features, extent, sizeTexture, style) {
-        // A texture is instancied drawn canvas
-        // origin and dimension are used to transform the feature's coordinates to canvas's space
-        const origin = new THREE.Vector2(extent.west(), extent.south());
-        const dimension = extent.dimensions();
-        const c = document.createElement('canvas');
-
-        c.width = sizeTexture;
-        c.height = sizeTexture;
-        const ctx = c.getContext('2d');
-        ctx.globalCompositeOperation = style.globalCompositeOperation || 'source-over';
-
-        const scale = new THREE.Vector2(ctx.canvas.width / dimension.x, ctx.canvas.width / dimension.y);
-
-        // Draw the canvas
-        if (Array.isArray(features)) {
-            for (let i = 0; i < features.length; i++) {
-                drawFeature(ctx, features[i], origin, scale, extent, style);
-            }
-        } else {
-            drawFeature(ctx, features, origin, scale, extent, style);
+/**
+ * Converts [Feature]{@link module:GeoJsonParser~Feature} to
+ * [THREE.Texture]{@link https://threejs.org/docs/#api/textures/Texture}.
+ *
+ * @function feature2Texture
+ *
+ * @param {module:GeoJsonParser~Feature} feature - a Feature or an array of
+ * Feature.
+ * @param {Object} [options] - Options controlling the conversion.
+ * @param {Extent} options.extent - The extent containing the feature.
+ * @param {number} options.size - The size of the texture, in pixels. The
+ * resulting texture will be a square, and it is highly recommended to use a
+ * power of 2 for GPU optimization purpose.
+ * @param {Object|function} options.style - The style to apply to the
```

Comment:
Oh I didn't realize functions were still supported. Which should actually gives all the possibilities I would ever want :-)